assets/js/components/Dashboard/NewPaMessage/StationsAndZones/SelectZonesPage.tsx

```diff
@@ -0,0 +1,541 @@
+          (s) =>
+            fp.intersection(routeToRouteIDMap[selectedRouteFilter], s.route_ids)
+              .length > 0,
+        );
+      }),
+    [filteredPlaces],
+  );
+
+  const screensByZone = useMemo(() => {
+    const screensWithAZone = allScreens.filter((s) => s.zone);
+    return fp.groupBy((s) => s.zone, screensWithAZone);
+  }, [allScreens]);
+
+  const areAllGroupsSelected = (groups: Array<ReadonlyArray<string>>) =>
+    groups.every((group) =>
+      group.every((zone) => {
+        return signIDs(screensByZone[zone] ?? []).every(isSelected);
+      }),
+    );
+
+  const selectGroups = (groups: Array<ReadonlyArray<string>>) => {
+    const signsWithoutSelectedRoute = fp.without(signIDs(allScreens), value);
+
+    const newScreens = groups.flatMap((group) =>
+      group.flatMap((zone) => signIDs(screensByZone[zone] ?? [])),
+    );
+
+    onChange(fp.uniq([...signsWithoutSelectedRoute, ...newScreens]));
+  };
+
+  const isAllSelected =
+    value.length > 0 &&
+    areAllGroupsSelected([LEFT_ZONES, MIDDLE_ZONES, RIGHT_ZONES]);
+  const isLeftSelected =
+    !isAllSelected && areAllGroupsSelected([LEFT_ZONES, MIDDLE_ZONES]);
+  const isRightSelected =
+    !isAllSelected && areAllGroupsSelected([MIDDLE_ZONES, RIGHT_ZONES]);
+
+  const hasRailFilter = () =>
+    [
+      "Green",
+      "Green-B",
+      "Green-C",
+      "Green-D",
+      "Green-E",
+      "Blue",
+      "Red",
+      "Orange",
+      "Mattapan",
+    ].includes(selectedRouteFilter);
+
+  const massSelectClassName = (selected: boolean) =>
+    selected ? "button-primary" : "button-secondary-outline";
+
+  const getFilterPillLabel = () => {
+    if (selectedRouteFilter.startsWith("Green-")) {
+      return `Green Line ${selectedRouteFilter.split("-")[1]}`;
+    } else {
+      return `${selectedRouteFilter} Line`;
+    }
+  };
+
+  return (
+    <div className="select-zones-page">
+      <div className="header">
+        <div className="title-and-edit">
+          <div className="title">Review Zones</div>
+          <div>
+            <Button
+              className="edit-button"
+              onClick={() => navigateTo(Page.STATIONS)}
+            >
+              Edit Stations
+            </Button>
+          </div>
+        </div>
+        <div className="buttons">
+          <Button
+            className="cancel-button"
+            onClick={() => navigateTo(Page.NEW)}
+          >
+            Cancel
+          </Button>
+          <Button
+            className="button-primary"
+            onClick={() => {
+              onSubmit(value);
+              navigateTo(Page.NEW);
+            }}
+          >
+            Done
+          </Button>
+        </div>
+      </div>
+      <div className="zone-selection">
+        <div className="filters-container">
+          <div>Service type</div>
+          <div className="filters">
+            {Object.keys(selectedRoutes).map((routeID) => {
+              let branchFilters;
```

Comment:
**suggestion:** no need for a mutable reference here. a ternary would allow this to be a `const` and colocate the value of `branchFilters` with its declaration.

```javascript
const branchFilters = routeID === "Green" ? selectedRoutes["Green"].sort().etc().etc() : null
```